src/algebra/module/pointwise_pi.lean

```diff
@@ -0,0 +1,88 @@
+/-
+Copyright (c) 2021 Alex J. Best. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex J. Best
+-/
+import algebra.pointwise
+import algebra.module.pi
+
+/-!
+# Pointwise actions on sets in Pi types
+
+This file contains lemmas about pointwise actions on sets in Pi types.
+
+## Tags
+
+set multiplication, set addition, pointwise addition, pointwise multiplication, pi
+
+-/
+
+open_locale pointwise
+open set
+
+@[to_additive]
+lemma smul_univ_pi {K R : Type*} [has_scalar K R] (ι : Type*) {r : K} (t : ι → set R) :
+  r • pi (univ : set ι) t = pi (univ : set ι) (λ (i : ι), r • t i) :=
```

Comment:
```suggestion
  r • pi (univ : set ι) t = pi (univ : set ι) (r • t) :=
```

Reply:
I'm not sure about this one, the version I had appears to be the simp normal form, so making this change I have to add a bunch of simps in my use case, I know this isn't a simp lemma, but surely the simp normal form is the one the library has chosen as preffered for future lemmas to match against?

Reply:
Lets come to a conclusion here before I start applying the other suggestions as all your proofs use this form. Thanks for the golfing though!

Reply:
Neither is the simp-normal form, as `pi.smul_def` is not marked `simp`. None of the proofs care about this form anyway, so I think you should apply all the suggestions and we can come back to removing these few characters.

Reply:
Sure, I'll apply the changes, and double check what changes I had to make in #2819 because of the suggested new signature